semver.go

```diff
@@ -173,16 +234,30 @@ func (v Version) Compare(o Version) int {
 
 }
 
+// IncrementPatch increments the patch version
+func (v *Version) IncrementRevision() error {
+	if v.Revision >= 0 {
+		v.Revision++
+	}
+	return nil
```

Comment:
So `-1` means revision is disabled and there should be no changes to revision if you call it; if there is a revision >=0 then increment. If revision is not parsed then we should make sure it never gets incremented or else it could get added to the string.